src/internal/components/ThemeEditor.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { LoadingScreen } from "../puck/components/LoadingScreen.tsx";
+import { InitialHistory, Config } from "@measured/puck";
+import { TemplateMetadata } from "../types/templateMetadata.ts";
+import { useMessageSenders } from "../hooks/useMessage.ts";
+import { useLocalStorage } from "../hooks/useLocalStorage.ts";
+import { DevLogger } from "../../utils/devLogger.ts";
+import { ThemeSaveState } from "../types/themeSaveState.ts";
+import { ThemeConfig } from "../../utils/themeResolver.ts";
+import { updateThemeInEditor } from "../../utils/applyTheme.ts";
+import { InternalThemeEditor } from "./InternalThemeEditor.tsx";
+
+const devLogger = new DevLogger();
+
+type ThemeEditorProps = {
+  puckConfig: Config;
+  templateMetadata: TemplateMetadata;
+  themeSaveState: ThemeSaveState | undefined;
+  themeData: any;
+  visualConfigurationData: any;
+  themeConfig: ThemeConfig | undefined;
+};
+
+export const ThemeEditor = (props: ThemeEditorProps) => {
+  const {
+    puckConfig,
+    templateMetadata,
+    themeSaveState,
+    themeData,
+    visualConfigurationData,
+    themeConfig,
+  } = props;
+
+  const {
+    sendDevThemeSaveStateData,
+    saveThemeSaveState,
+    publishThemeConfiguration,
+    deleteThemeSaveState,
+    sendDevLayoutSaveStateData,
+  } = useMessageSenders();
+
+  const { buildThemeLocalStorageKey, clearThemeLocalStorage } =
+    useLocalStorage(templateMetadata);
+
+  const [puckInitialHistory, setPuckInitialHistory] = useState<
+    InitialHistory | undefined
+  >();
+  const [puckInitialHistoryFetched, setPuckInitialHistoryFetched] =
+    useState<boolean>(false);
+  const [themeInitialHistory, setThemeInitialHistory] =
+    useState<ThemeSaveState>({ history: [{}], index: 0 });
+  const [themeInitialHistoryFetched, setThemeInitialHistoryFetched] =
+    useState<boolean>(false);
+
+  /**
+   * Determines the initialHistory to send to Puck.
+   * For theme mode, only load from Content
+   */
+  const loadPuckInitialHistory = useCallback(() => {
+    if (!templateMetadata) {
+      return;
+    }
+    devLogger.logFunc("loadPuckInitialHistory");
+
+    // Only load Content data for theme mode
+    if (templateMetadata.isThemeMode) {
```

Comment:
Presumably we've already guarded the component by this so any isThemeMode checks should be unncessary.